src/XCCDF_POLICY/xccdf_policy.c

```diff
@@ -981,6 +979,52 @@ _xccdf_policy_rule_evaluate(struct xccdf_policy * policy, const struct xccdf_rul
 		return _xccdf_policy_report_rule_result(policy, result, rule, NULL, XCCDF_RESULT_NOT_SELECTED, NULL);
 	}
 
+	bool all_requires_met = true;
```

Comment:
@yuumasato each profile is resolved before it can be evaluated. We have all the information we need to select or unselect things at resolve stage. Can you elaborate why this can't be done at resolve stage? You probably didn't reach xccdf_resolve_* because your benchmark was already resolved.